src/Tribe/Log/Admin.php

```diff
@@ -0,0 +1,249 @@
+	 */
+	protected function setup_script() {
+		wp_enqueue_script(
+			'tribe-common-logging-controls',
+			tribe_resource_url( 'admin-log-controls.js', false, 'common' ),
+			array( 'jquery' ),
+			Tribe__Main::VERSION,
+			true
+		);
+
+		wp_localize_script( 'tribe-common-logging-controls', 'tribe_logger_data', array(
+			'check' => wp_create_nonce( 'logging-controls' )
+		) );
+	}
+
+	/**
+	 * Returns a list of logs that are available for perusal.
+	 *
+	 * @return array
+	 */
+	protected function get_available_logs() {
+		$available_logs = $this->current_logger()->list_available_logs();
+
+		if ( empty( $available_logs ) ) {
+			return array( '' => _x( 'None currently available', 'log selector', 'tribe-common' ) );
+		}
+
+		return $available_logs;
+	}
+
+	/**
+	 * Returns a list of logging engines that are available for use.
+	 *
+	 * @return array
+	 */
+	protected function get_log_engines(){
+		$available_engines = $this->log_manager()->get_logging_engines();
+
+		if ( empty( $available_engines ) ) {
+			return array( '' => _x( 'None currently available', 'log engines', 'tribe-common' ) );
+		}
+
+		$engine_list = array();
+
+		foreach ( $available_engines as $class_name => $engine ) {
+			/**
+			 * @var Tribe__Log__Logger $engine
+			 */
+			$engine_list[ $class_name ] = $engine->get_name();
+		}
+
+		return $engine_list;
+	}
+
+	/**
+	 * Returns all log entries for the current or specified log.
+	 *
+	 * @return array
+	 */
+	protected function get_log_entries( $log = null ) {
+		$logger = $this->current_logger();
+		$logger->use_log( $log );
+		return $logger->retrieve();
+	}
+
+	/**
+	 * Returns an array of logging levels arranged as key:value pairs, with
+	 * each key being the level code and the value being the human-friendly
+	 * description.
+	 *
+	 * @return array
+	 */
+	protected function get_logging_levels() {
+		$levels = array();
+		$available_levels = $this->log_manager()->get_logging_levels();
+
+		foreach ( $available_levels as $logging_level ) {
+			$levels[ $logging_level[0] ] = $logging_level[1];
+		}
+
+		return $levels;
+	}
+
+	/**
+	 * Provides a URL that can be used to download the current or specified
+	 * log.
+	 *
+	 * @param $log
+	 *
+	 * @return string
+	 */
+	protected function get_log_url( $log = null ) {
+		$query = array(
+			'tribe-common-log' => 'download',
+			'check' => wp_create_nonce( 'download_log' )
+		);
+
+		$log_download_url = add_query_arg( $query, get_admin_url( null, 'edit.php' ) );
+
+		return esc_url( $log_download_url );
+	}
+
+	/**
+	 * Facilitate downloading of logs.
+	 */
+	public function serve_log_downloads() {
+		if ( empty( $_GET['tribe-common-log'] ) || 'download' !== $_GET['tribe-common-log'] ) {
+			return;
+		}
+
+		if ( ! wp_verify_nonce( @$_GET['check'], 'download_log' ) ) {
+			return;
+		}
+
+		if ( ! empty( $_GET['log'] ) && in_array( $_GET['log'], $this->get_available_logs() ) ) {
+			$log_name = filter_var( $_GET['log'], FILTER_SANITIZE_STRING, FILTER_FLAG_STRIP_HIGH );
```

Comment:
shouldn't we use? [`sanitize_text_field`](https://codex.wordpress.org/Function_Reference/sanitize_text_field)